### PR TITLE
Fix: Only call method channels on native platforms

### LINKED ISSUE #1176: `MethodChannel` is called for platforms that dont have some implementations such as `endNativeFrames`
### Description

For example https://github.com/getsentry/sentry-dart/blob/d235a79dc30772bb861de3eec8222cf2300bbd23/flutter/lib/src/sentry_native_channel.dart#L39-L44
On the Web, `endNativeFrames` returns an empty string so it fails to cast to `<String, dynamic>`.
Since we swallow its exceptions, it's not a problem, but we should not call such methods in the first place, only for Android, iOS, and macOS for now.

### PR DESCRIPTION
**Pull request by denrase**
## :scroll: Description

Closes #1176

## :bulb: Motivation and Context

- Only call native cannels on native platforms
- Also call native scope observers on iOS & macOS

## :green_heart: How did you test it?

Unit tests.

## :pencil: Checklist

- [x] I reviewed submitted code
- [x] I added tests to verify changes
- [ ] I updated the docs if needed
- [x] All tests passing
- [x] No breaking changes

### CODE REVIEW

File: flutter/test/sentry_flutter_util.dart [hunk 1]
```diff
@@ -13,6 +14,18 @@ void testTransport({
...
+void testScopeObserver(
+    {required SentryFlutterOptions options,
+    required bool hasNativeScopeObserver}) {
+  try {
+    options.scopeObservers
+        .firstWhere((element) => element.runtimeType == NativeScopeObserver);
+    expect(true, hasNativeScopeObserver);
+  } catch (_) {
+    expect(false, hasNativeScopeObserver);
+  }
```
Comment (marandaneto) [inline flutter/test/sentry_flutter_util.dart]: Do we actually need a try catch?
If it throws, means the test failed, and this is what we expect.

**Review by marandaneto — APPROVED** [review]
Left a [comment](https://github.com/getsentry/sentry-dart/pull/1196/files#r1053085976) otherwise LGTM.

File: flutter/test/sentry_flutter_util.dart (hunk 1 above)
Comment (denrase) [reply flutter/test/sentry_flutter_util.dart]: We'd have to do the catch on the test side, as having `hasNativeScopeObserver `, false is also a valid thing we want to test and not have the code flow interrupted by an exception. I tried using the non-throwing `firstWhereOrNull`, but it is emitting a warning when importing the needed collection dependency.

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,9 @@
 
 ## Unreleased
 
+### Fixes
+
+- Fix: Only call method channels on native platforms ([#1196](https://github.com/getsentry/sentry-dart/pull/1196))
```
Comment (marandaneto) [inline CHANGELOG.md:7]:
```suggestion
### Fixes

- Fix: Only call method channels on native platforms ([#1196](https://github.com/getsentry/sentry-dart/pull/1196))

```

File: flutter/test/sentry_flutter_util.dart (hunk 1 above)
Comment (marandaneto) [reply flutter/test/sentry_flutter_util.dart]: Not sure I understand this correctly, if it's a boolean, we can assert true or false without catching errors, maybe the solution is to not use `firstWhere` but rather something that returns null instead, so it won't throw?

**Comment by marandaneto** [comment]
@denrase have you checked that the scope sync is working for iOS/macOS now? we need to be sure before going ahead.

File: flutter/test/sentry_flutter_util.dart (hunk 1 above)
Comment (denrase) [reply flutter/test/sentry_flutter_util.dart]: Refctored so we don't rely on the throw.

**Comment by marandaneto** [comment]
> @denrase have you checked that the scope sync is working for iOS/macOS now? we need to be sure before going ahead.

@denrase is this tested? ready to be merged?

**Comment by denrase** [comment]
@marandaneto Set a sample user through scope sync in the example app and let it crash in swift code, user was synced. On the `main` branch, there is no user with the same code.

**Here**

Bildschirmfoto 2023-01-09 um 12 55 40

**Main**

Bildschirmfoto 2023-01-09 um 13 04 21

**Comment by denrase** [comment]
@marandaneto Also works on macOS.

Bildschirmfoto 2023-01-09 um 13 14 41

**Comment by denrase** [comment]
Tested with all methods. Issue with `removeContexts` and `removeTag`, which may be an [issue in the cocoa SDK.](https://github.com/getsentry/sentry-cocoa/issues/2588)

**Pull request merged**